code/drasil-data/Data/Drasil/SentenceStructures.hs

```diff
@@ -233,20 +233,15 @@ constraintToExpr c (EnumeratedStr _ l) = isin (sy c) (DiscreteS l)
 
 --Formatters for the constraints
 fmtPhys :: (Constrained c, Quantity c) => c -> Sentence
-fmtPhys c = foldIfNotEmpty c $ filter isPhysC (c ^. constraints)
+fmtPhys c = foldConstraints c $ filter isPhysC (c ^. constraints)
 
 fmtSfwr :: (Constrained c, Quantity c) => c -> Sentence
-fmtSfwr c = foldIfNotEmpty c $ filter isSfwrC (c ^. constraints)
+fmtSfwr c = foldConstraints c $ filter isSfwrC (c ^. constraints)
 
 -- Helper for formatting constraints
-foldIfNotEmpty :: (Quantity c) => c -> [Constraint] -> Sentence
-foldIfNotEmpty _ [] = EmptyS
-foldIfNotEmpty c e  = E $ foldConstraints $ map (constraintToExpr c) e
-
-foldConstraints :: [Expr] -> Expr
-foldConstraints []     = error "foldIfNotEmpty didn't catch an empty list of constraints."
-foldConstraints [x]    = x
-foldConstraints (x:xs) = x $&& foldConstraints xs
+foldConstraints :: (Quantity c) => c -> [Constraint] -> Sentence
+foldConstraints _ [] = EmptyS
+foldConstraints c e  = E $ foldl1 ($&&) $ map (constraintToExpr c) e
```

Comment:
`intersperse` isn't applicable anymore when using the `($&&)` operator on the expressions (since I'm no longer hardcoding `S "∧"` as a `Sentence`), but I think this is a nice compromise. @JacquesCarette

Reply:
Agreed.